File: gdscript/src/main/java/gdscript/psi/impl/GdBitAndExImpl.java
package gdscript.psi.impl;

import com.intellij.lang.ASTNode;
import com.intellij.psi.PsiElementVisitor;
import com.intellij.psi.util.PsiTreeUtil;
import gdscript.psi.GdBitAndEx;
import gdscript.psi.GdBitAndSign;
import gdscript.psi.GdExpr;
import gdscript.psi.GdVisitor;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public class GdBitAndExImpl extends GdExprImpl implements GdBitAndEx {

  public GdBitAndExImpl(@NotNull ASTNode node) {
    super(node);
  }

  @Override
  public void accept(@NotNull GdVisitor visitor) {
    visitor.visitBitAndEx(this);
  }

  @Override
  public void accept(@NotNull PsiElementVisitor visitor) {
    if (visitor instanceof GdVisitor) accept((GdVisitor)visitor);
    else super.accept(visitor);
  }

  @Override
  @NotNull
  public GdBitAndSign getBitAndSign() {
    return notNullChild(PsiTreeUtil.getChildOfType(this, GdBitAndSign.class));
  }

  @Override
  @NotNull
  public List<GdExpr> getExprList() {
    return PsiTreeUtil.getChildrenOfTypeAsList(this, GdExpr.class);
  }

}

File: gdscript/src/main/java/gdscript/psi/impl/GdBitAndExImpl.kt
package gdscript.psi.impl

import com.intellij.lang.ASTNode
import com.intellij.psi.PsiElementVisitor
import com.intellij.psi.util.PsiTreeUtil
import gdscript.psi.GdBitAndEx
import gdscript.psi.GdBitAndSign
import gdscript.psi.GdExpr
import gdscript.psi.GdVisitor

class GdBitAndExImpl(node: ASTNode) : GdExprImpl(node), GdBitAndEx {
    override fun accept(visitor: GdVisitor) {
        visitor.visitBitAndEx(this)
    }

    override fun accept(visitor: PsiElementVisitor) {
        if (visitor is GdVisitor) accept(visitor)
        else super.accept(visitor)
    }

    override val bitAndSign: GdBitAndSign
        get() = notNullChild(PsiTreeUtil.getChildOfType(this, GdBitAndSign::class.java))

    override val exprList: List<GdExpr>
        get() = PsiTreeUtil.getChildrenOfTypeAsList(this, GdExpr::class.java)
}
